subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/XCTestIntegrationTest.groovy

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest
+
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.SwiftXcTestTestApp
+import org.gradle.nativeplatform.fixtures.app.TestElement
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+
+import static org.gradle.nativeplatform.fixtures.app.SourceTestElement.newTestSuite
+import static org.gradle.nativeplatform.fixtures.app.SourceTestElement.newTestCase
+
+@Requires([TestPrecondition.SWIFT_SUPPORT, TestPrecondition.MAC_OS_X])
+class XCTestIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
```

Comment:
we should add some coverage for:

- What happens when I apply the xctest plugin along with one of the C++ plugins and run 'gradle test'? What happens if I do this and have some C++ main source and some Swift test source code?
- What happens when I apply the xctest plugin without applying the Swift library or executable plugin and run `gradle test`? What happens if I do this and have some Swift test source code?
- That `gradle assemble` doesn't build or run any of the tests.